File: CPP/Stack.cpp
#include <iostream>
using namespace std;

struct node
{
    int value;
    node *next;
};

node *createNode(int value)
{
    node *n1 = new node;
    n1->value = value;
    n1->next = NULL;
    return n1;
}

class Stack
{
    node *top;

public:
    Stack() { top = NULL; }
    void push(int value)
    {
        node *n1 = createNode(value);
        if (top == NULL)
        {
            top = n1;
            return;
        }
        n1->next = top;
        top = n1;
    }
    void pop()
    {
        if (top == NULL)
        {
            cout << "UnderFlow\n";
            return;
        }
        node *temp = top;
        top = top->next;
        free(temp);
    }
    void displayAll()
    {
        if (top == NULL)
        {
            cout << "UnderFlow\n";
            return;
        }
        node *ptr = top;
        while (ptr != NULL)
        {
            cout << ptr->value << "->";
            ptr = ptr->next;
        }
        cout << endl;
    }
};

int main()
{
    Stack s;
    s.push(1);
    s.push(2);
    s.push(3);
    s.displayAll();
    s.pop();
    s.displayAll();
    return 0;
}

File: Python/Data_Structures/Stack.py
class node(object):
    def __init__(self, value = None):
        self.value = value
        self.next = None


class Stack(object):
    def __init__(self, value = None):
        self.top = node(value)
    
    def push(self, value):
        n1 = node(value)
        if self.top == None:
            self.top = n1
            return
        n1.next = self.top
        self.top = n1

    def pop(self):
        if self.top == None:
            raise ValueError('Underflow')
        temp = self.top.value
        self.top = self.top.next
        return temp

    def displayAll(self):
        if self.top == None:
            raise ValueError('Underflow')
        ptr = self.top
        while ptr != None:
            print(ptr.value, end=' ')
            ptr = ptr.next
        print()

# testing code

S = Stack(1)

S.push(2)
S.push(3)
S.displayAll()
S.push(4)
S.push(5)
S.displayAll()
S.pop()
S.pop()
S.displayAll()

